pilot/pkg/serviceregistry/kube/conversion.go

```diff
@@ -183,11 +198,30 @@ func ExternalNameServiceInstances(k8sSvc *coreV1.Service, svc *model.Service) []
 	return out
 }
 
+// NamespacedNameForK8sObject is a helper that creates a NamespacedName for the given K8s Object.
+func NamespacedNameForK8sObject(obj metav1.Object) types.NamespacedName {
+	return types.NamespacedName{
+		Namespace: obj.GetNamespace(),
+		Name:      obj.GetName(),
+	}
+}
+
 // ServiceHostname produces FQDN for a k8s service
+// TODO(nmittler): Rename this to ServiceClusterLocalHostname to differentiate with ServiceClusterSetLocalHostname.
 func ServiceHostname(name, namespace, domainSuffix string) host.Name {
 	return host.Name(name + "." + namespace + "." + "svc" + "." + domainSuffix) // Format: "%s.%s.svc.%s"
 }
 
+// ServiceHostnameForKR calls ServiceHostname with the name and namespace of the given kubernetes resource.
+func ServiceHostnameForKR(obj metav1.Object, domainSuffix string) host.Name {
+	return ServiceHostname(obj.GetName(), obj.GetNamespace(), domainSuffix)
+}
+
+// ServiceClusterSetLocalHostname produces Kubernetes Multi-Cluster Services (MCS) ClusterSet FQDN for a k8s service
+func ServiceClusterSetLocalHostname(nn types.NamespacedName) host.Name {
```

Comment:
NamespacedName causes an extra conversion

Reply:
Yes, but it's pretty minor and significantly reduces chances of accidental misuse of the method when passing in two strings.